i18n/ja/docusaurus-plugin-content-docs/current/guides/integration-dsp-no-sdk.md

```diff
@@ -0,0 +1,123 @@
+
+To review detailed logic, see [UID2Encryption.cs, lines 36-50](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/UID2Encryption.cs#L36-L50).
+
+### Decrypt the Token
+
+Use the master key and site key to decrypt the token. For a code example, refer to the `Decrypt` function: see [UID2Encryption.cs, line 29](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/UID2Encryption.cs#L29). This function decrypts UID2 tokens into raw UID2s as part of the UID2 SDK for C# / .NET, and includes logic to handle different token versions.
+
+### Make Sure the Token Has Not Expired
+
+A token must be valid and current so that it can be used in the bidstream. You must make sure that it hasn't expired. To do this, check these two conditions:
+
+- The `remaining_lifetime` or `lifetime` value must not exceed the `max_bidstream_lifetime_seconds` value.
+- The `skew_duration` value must not exceed the `allow_clock_skew_seconds` value.
+
+ For an example of how this is done, review the code for the `DoesTokenHaveValidLifetimeImpl` function: see [UID2Encryption.cs, line 237](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/UID2Encryption.cs#L237).
+
+The following sections show how the `lifetime` and `skew_duration` values for a token are calculated. The calculation is a little different depending on the token version:
+
+- [Calculating Token Expiration: Token v2](#calculating-token-expiration-token-v2)
+- [Calculating Token Expiration: All Later Versions](#calculating-token-expiration-all-later-versions)
+
+#### Calculating Token Expiration: Token v2
+
+For token v2, the calculation to make sure that the token hasn't expired is as follows:
+
+`remaining_lifetime` = **Token Expiry** &#8211; **Current Time**
+
+`skew_duration` = `0`
+
+For v2, we use the token expiry minus the current time to calculate the remaining lifetime. This is because v2 doesn't have a **Token Generated** field, which is present in later versions. Instead, it has an **Identity Established** field: this indicates the time that the original token was generated, before any token refreshes, so it can't be used to calculate whether the token is still valid.
+
+#### Calculating Token Expiration: All Later Versions
+
+For all token versions later than v2, the calculation to make sure that the token hasn't expired is as follows:
+
+`lifetime` = **Token Expiry** &#8211; **Token Generated**
+
+`skew_duration` = **Token Generated** &#8211; **Current Time**
+
+Again, it's important to know that versions later than v2 have a **Token Generated** field. If the token was refreshed, this value is updated, so you can use it to calculate the token lifetime. The **Token Generated** field in v2 doesn't get updated when the token is refreshed, so it isn't useful for this calculation.
```

Comment:
Actually v2 does not have a `Token Generated ` field, as what is mentioned in L107

Reply:
Yes thx! This was a typo. Good catch. I meant to refer to the Identity Established field. But I think it isn't necessary to mention the Identity Established field here. I simplified to this:

Versions later than v2 have a Token Generated field, which is updated if the token is refreshed, so we use this to calculate the token lifetime.